pytest_girder/pytest_girder/fixtures.py

```diff
@@ -130,9 +138,40 @@ def server(db, request):
         cherrypy.engine.stop()
         cherrypy.engine.exit()
         cherrypy.tree.apps = {}
+        # This is needed to allow cherrypy to restart on another port
+        cherrypy.server.httpserver = None
         docs.routes.clear()
 
 
+@pytest.fixture
+def server(db, request):
+    """
+    Require a CherryPy embedded server.
+
+    Provides a started CherryPy embedded server with a request method for
+    performing local requests against it. Note: this fixture requires the db
+    fixture.
+    """
+    for server in _yieldServer(request):
+        yield server
```

Comment:
Is the loop necessary, doesn't `_yieldServer` only yield a single thing?

Reply:
Somehow, pytest doesn't actually do things correctly unless the pytest.fixture uses the yield keyword itself.  Pythonically, `return _yieldServer(request)` works, but pytest doesn't work for me when I do that.  Maybe I'm missing something.

Reply:
What about `yield _yieldServer(request)`?

Reply:
That returns a generator of a generator not a generator itself.

Reply:
And `next(_yieldServer(request))` never calls the cleanup code of _yieldserver since it doesn't ask for the final (non existent) element.

Reply:
Hm, carry on then!

Reply:
One could do:
```
server = _yieldServer(request)
yield next(server)
next(server)
```